php/classes/integrations/elementor/class-template-importer.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace SeriouslySimplePodcasting\Integrations\Elementor;
+
+use Elementor\TemplateLibrary\Manager;
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Template Importer for Elementor
+ *
+ * @author      Jonathan Bossenger
+ * @category    Class
+ * @package     SeriouslySimplePodcasting/Classes
+ * @since       1.19.18
+ */
+class Template_Importer {
+
+	protected $template_name = 'ssp-elementor-template-2020-09-30.json';
+	protected $template_path = SSP_PLUGIN_PATH . 'templates/elementor/';
+
+	protected $template_library_manager;
+
+	public function __construct() {
+		$this->template_library_manager = new Manager();
+		$this->init();
+	}
+
+	public function init() {
+		add_action( 'init', array( $this, 'process_template_import' ) );
+	}
+
+	public function process_template_import() {
+		// @todo add is_admin checking
+
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		// @todo add user caps checks
+		// @todo add nonce checking
+		if ( ! isset( $_GET['elementor_import_templates'] ) ) {
+			return;
+		}
+
+		if ( ! $_GET['elementor_import_templates'] == true ) {
+			return;
+		}
+
+		if ( $_GET['tab'] != 'extensions' ) {
+			return;
+		}
+
+		// verify template import nonce
+		if ( ! isset( $_GET['import_template_nonce'] ) || ! wp_verify_nonce( $_GET['import_template_nonce'], '' ) ) {
```

Comment:
I wouldn't trigger an admin notice here, I would just return;

The reason for this is that the admin notice would show someone that a security check has failed, which might alert someone to the fact that there is a security check in place.